pallets/settlement/src/lib.rs

```diff
@@ -742,9 +626,7 @@ decl_module! {
         /// # Arguments
         /// * `instruction_id` - Instruction id to reject.
         /// * `portfolios` - Portfolios that the sender controls and wants them to reject this instruction
-        #[weight = weight_for::weight_for_reject_instruction::<T>()
-            + weight_for::weight_for_transfer::<T>() // Maximum weight for `execute_instruction()`
-        ]
+        #[weight = <T as Trait>::WeightInfo::reject_instruction_with_no_pre_affirmations(portfolios.len() as u32)]
         pub fn reject_instruction(origin, instruction_id: u64, portfolios: Vec<PortfolioId>) -> DispatchResult {
```

Comment:
How are you limiting legs here? One portfolio can be involved in multiple legs and the weight should be a function of the number of legs, not the number of portfolios.

Reply:
Same as above

Reply:
(Closing, let's continue the discussion at a single place)